mitre_fast_layered_map_obstacle_filters:

  - name: bayes_update
    type: mitreFastLayeredMap/BayesUpdate
    params:
      probability_layer: current_probability
      history_layer_prefix: "history_"
      history_count: 5
      # Advanced settings for sensor model
      starting_prob: 0.5
      prob_sense_emp_given_occ: 0.2
      prob_sense_emp_given_emp: 0.9
      prob_sense_occ_given_occ_rate: 0.1
      prob_sense_occ_given_occ_offset: 0.3
      prob_sense_occ_given_emp_rate: -0.1
      prob_sense_occ_given_emp_offset: 0.5

  - name: threshold_filter_nonground
    type: mitreFastLayeredMap/ThresholdFilter
    params:
      value_layer: current_probability
      output_layer: nonground
      threshold: 0.8
      output_value: 100
  
mitre_fast_layered_map_map_operations:
  - name: ray_trace_2d
    type: mitreFastLayeredMap/RayTrace2d
    params:
      nonground_layer: nonground
      ground_layer: ground

  - name: inflation
    type: mitreFastLayeredMap/Inflation
    params:
      layer: nonground
      inflation_side_len_m: 1